python/lsst/pipe/supertask/super_task.py

```diff
@@ -318,3 +319,72 @@ def _makeArgumentParser(cls):
         parser = ArgumentParser(name=task_name)
         parser.add_id_argument("--id", "raw", help="data IDs, e.g. --id visit=12345 ccd=1,2^0,3")
         return parser
+
+    def write_config(self, butler, clobber=False, do_backup=True):
+        """!Write the configuration used for processing the data, or check that an existing
+
+        one is equal to the new one if present.
+
+        @param[in] butler   data butler used to write the config.
+            The config is written to dataset type self._get_config_name()
+        @param[in] clobber  a boolean flag that controls what happens if a config already has been saved:
+            - True: overwrite the existing config
+            - False: raise TaskError if this config does not match the existing config
+        """
+        config_name = self._get_config_name()
+        if config_name is None:
+            return
+        if clobber:
+            butler.put(self.config, config_name, doBackup=do_backup)
+        elif butler.datasetExists(config_name):
+            # this may be subject to a race condition; see #2789
+            try:
+                old_config = butler.get(config_name, immediate=True)
+            except Exception as exc:
+                raise type(exc)("Unable to read stored config file %s (%s); consider using --clobber-config" %
+                                (config_name, exc))
+            output = lambda msg: self.log.fatal("Comparing configuration: " + msg)
```

Comment:
I copied this directly from `pipe.base.cmdLineTask`, so prefer not to change things.